graph/number_of_islands_dfs.cpp

```diff
@@ -0,0 +1,103 @@
+/**
+ * @file
+ *
+ * @brief Algorithm to count the number of islands
+ *
+ * @details
+ * Depth First Traversal (or Search) for a graph is similar to Depth First
+ * Traversal of a tree. The only catch here is, that, unlike trees, graphs may
+ * contain cycles (a node may be visited twice). To avoid processing a node more
+ * than once, use a boolean visited array. A graph can have more than one DFS
+ * traversal.
+ *
+ * The algorithm implemented in this file determines the number of islands in a
+ * 2D array of 1s(land) and 0s(water).
+ *
+ * @author [Harsh Agrawal](https://github.com/harshagrawal523)
+ *
+ */
+#include <iostream>
+#include <string>
+#include <vector>
```

Comment:
Please add a one-line description of what the library/header is for (**see the example below**).

```c++
#include <iostream>    /// for IO operations
#include <cassert>    /// for assert
```